src/bundle/DependencyInjection/Configuration.php

```diff
@@ -93,74 +93,94 @@ public function getConfigTreeBuilder()
+                    ->arrayNode('user')
+                        ->children()
+                            ->scalarNode('endpoint')
+                                ->info('User API endpoint')
+                                ->example('end_point: https://user.net')
+                            ->end()
                         ->end()
                     ->end()
                 ->end()
-                ->arrayNode('field')
-                    ->children()
-                        ->arrayNode('identifiers')
-                            ->children()
-                                ->arrayNode('intro')
-                                    ->info('Content type intro field')
-                                    ->example('blog_post: blog_post_intro_field')
-                                    ->scalarPrototype()->end()
-                                ->end()
-                                ->arrayNode('image')
-                                    ->info('Content type image field')
-                                    ->example('article: article_image_field')
-                                    ->scalarPrototype()->end()
-                                ->end()
-                                ->arrayNode('author')
-                                    ->info('Content type author field')
-                                    ->example('article: article_author_field')
-                                    ->scalarPrototype()->end()
-                                ->end()
+            ->end()
+            ->arrayNode('field')
+                ->children()
+                    ->arrayNode('identifiers')
+                        ->children()
+                            ->arrayNode('intro')
+                                ->info('Content type intro field')
+                                ->example('blog_post: blog_post_intro_field')
+                                ->scalarPrototype()->end()
+                            ->end()
+                            ->arrayNode('image')
+                                ->info('Content type image field')
+                                ->example('article: article_image_field')
+                                ->scalarPrototype()->end()
+                            ->end()
+                            ->arrayNode('author')
+                                ->info('Content type author field')
+                                ->example('article: article_author_field')
+                                ->scalarPrototype()->end()
                             ->end()
                         ->end()
-                        ->arrayNode('relations')
-                        ->end()
+                    ->end()
+                    ->arrayNode('relations')
+                        ->beforeNormalization()
+                            ->ifTrue(
+                                function (array $relations): bool {
+                                    $pattern = '/\D+\.\D+/';
```

Comment:
Shouldn't this be:

```suggestion
                                    $pattern = '/^\D+\.\D+$/';
```

Otherwise you're only checking that the string contains `"not_digit.not_digit" string. What I believe we're looking for is a strict check that it is strictly that string.

Example of a string that would pass the previous check: `"content.content_type.field"`

Reply:
After deeply investigation it seems that unfortunately regex will not properly work if we based only on not digit characters. 

What about pattern like this: `^(\b[a-zA-Z][\w]+)\.(\b[a-zA-Z][\w]+)$`? 

Here you can see the demo: https://regexr.com/5q77r

Reply:
Looks a lot better. You can use just 

```diff
- ^(\b[a-zA-Z][\w]+)\.(\b[a-zA-Z][\w]+)$
+ ^[a-zA-Z][\w]+\.[a-zA-Z][\w]+$
```

No need for capture groups (`()`) nor word boundary checks (`\b`).